azurerm/resource_arm_mariadb_database_test.go

```diff
@@ -0,0 +1,133 @@
+package azurerm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMMariaDbDatabase_basic(t *testing.T) {
+	resourceName := "azurerm_mariadb_database.test"
+	ri := acctest.RandInt()
+	config := testAccAzureRMMariaDbDatabase_basic(ri, testLocation())
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMMariaDbDatabaseDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMMariaDbDatabaseExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "charset", "utf8"),
+					resource.TestCheckResourceAttr(resourceName, "collation", "utf8_general_ci"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMMariaDbDatabaseExists(name string) resource.TestCheckFunc {
```

Comment:
Could we use `resourceName` here to prevent conflict with name below?

Reply:
ARGH! I did it again. lol... good catch!